numbat/src/ast.rs

```diff
@@ -58,7 +58,11 @@ impl PrettyPrint for BinaryOperator {
 #[derive(Debug, Clone, PartialEq)]
 pub enum StringPart {
     Fixed(String),
-    Interpolation(Span, Box<Expression>),
+    Interpolation {
+        span: Span,
+        expr: Box<Expression>,
+        specifiers: String,
```

Comment:
I think I would prefer `Option<String>` to clearly differentiate between an empty and a missing format specifier.